### ISSUE TITLE
wiki add file upload

### ISSUE THREAD

**Issue by BruinY** [gogits/gogs#2263]

**Comment by Unknwon** [comment]
Thanks your feedback!

But why?

**Comment by bkcsoft** [comment]
Most likely for images. I'd also like this feature ;)

**Comment by Unknwon** [comment]
ping @BruinY

**Comment by biji** [comment]
Yes, new feature to upload images to __wiki__ would be great. just like adding images to __Issue__

Currently I push images to master repository , then permalink it to __wiki__ pages

**Comment by Unknwon** [comment]
Thanks @biji .

Any suggestions to solve this problem will be appreciate!

**Comment by sapk** [comment]
Normally it should be possible to clone the wiki and directly add image via git. I think it is how github permit it. 
https://help.github.com/articles/adding-images-to-wikis/
http://mikehadlow.blogspot.fr/2014/03/how-to-add-images-to-github-wiki.html

**Comment by Unknwon** [comment]
I think directly use images in the repository is the best way.

**Comment by biji** [comment]
I cant find raw premalink for images when adding image to cloned wiki. But if I add image to repository, I can find raw permalink to add to wiki

**Comment by Unknwon** [comment]
@biji yes, we need a way/link similar to GitHub that allows you access the raw image.

**Comment by galaktor** [comment]
I was trying to do this same thing. What I expected to be able to do is add images to the `<repo>.wiki.git` repository and reference them with relative paths in markdown. I tried various permutations of *raw* and *blob* links but to no avail. I wouldn't want to host the images for the wiki in a separate repository.

**Comment by Unknwon** [comment]
@galaktor sorry for the trouble, but wiki repository does not handle anything other than `md` files...

**Comment by galaktor** [comment]
No problem, I figured that much. That said, I did push an image to the `.wiki` repo, but just wasn't able to reference it via markdown URL. It would be brilliant to have wiki + assets inside the same wiki repo. Technically the wiki is incomplete without it's attachments.

**Comment by blinzelaffe** [comment]
+1

**Comment by sfkleach** [comment]
+1

**Comment by Vandewaetere** [comment]
+1 for easier image upload (currently doing permalink from repo)

**Comment by javydreamercsw** [comment]
This would be an useful feature +1!

**Comment by javydreamercsw** [comment]
@Vandewaetere any hints on how to do the permalink?

**Comment by bobymicroby** [comment]
+1

**Comment by sfkleach** [comment]
+1

**Comment by jorise7** [comment]
+1

**Comment by baikangwang** [comment]
+1

**Comment by Fastidious** [comment]
The wiki, in general, needs some love (see #3842). I join this handsome group of fellows on the desire to have the feature requested here as well.

**Comment by sh2ka** [comment]
+1. I wonder why this was not implemented yet, especially when "issues" already have this feature?

**Comment by DMWOO7** [comment]
I see an additional value in this feature, too. Not only for images, which are common in documentations: Using attachments, we could also add files, that are related to the project. For example, we've external ressources like documentation files, that are related to the project. Currently our workaround is to insert them in git by adding a second gogs project for those documentation files. Would be nice if we could attach them in the wiki, so we've a single project for all of them and are able to access them using the gogs web UI.

**Comment by DeanPDX** [comment]
This would be really helpful for me as well! I'm trying not to clog my main repo up by adding images to it but I don't see a way around it.

**Comment by bigornoo** [comment]
+1

**Comment by motu81** [comment]
I tried to add a image via git to the *.wiki.git repo, that is possible but it is never served, so it is unuseable in the wiki pages.

**Comment by AlgyTaylor** [comment]
+1 for this.  I'd be perfectly happy if I could just add an image to the *.wiki.git repository, upload it, then be able to link to it through the UI (e.g. ``https://my.gogs.location/algy.taylor/repo/wiki/uploaded-image.png``).

**Comment by jcolomb** [comment]
coming back to this, as it would be great. In my case I want the wiki public and the repo private, so adding images to the main repo does not work.

**Comment by brianwestphal** [comment]
also less-technical project managers probably can't deal with having to clone the wiki locally
